google/cloud/spanner/sql_statement.cc

```diff
@@ -0,0 +1,40 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/sql_statement.h"
+#include <algorithm>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+bool operator==(SqlStatement const& lhs, SqlStatement const& rhs) {
+  std::string s1 = lhs.statement_;
```

Comment:
Hmmm, I think `SELECT * FROM my_table WHERE foo='a capitalized string'` is a different statement than `SELECT * FROM my_table WHERE foo='A Capitalized String'`.  I know the keywords and functions are case-insensitive, but I sure the string values are not, and and I am not sure about table names and such.

Reply:
That's a good point. Perhaps we should just punt on the case insensitivity of the keywords and stuff and just do a dead-simple case *sensitive* comparison here? I mean, in the C++ realm, "foo" != "FOO", so it may not be that surprising of SqlStatement behaves the same way for its stringy parts.